Automattic/jetpack
# My Jetpack: Add Boost speed score into boost product card.

This PR adds the site's Boost speed score into the My Jetpack Boost product card. (which adds more value to the Boost card, see related project ⤵️)

Related to: [Add value to the active Boost product card on My Jetpack](https://github.com/orgs/Automattic/projects/702/views/6?pane=issue&itemId=50279848)





**Screenshots:**

![Markup 2024-02-12 at 15 49 59](https://github.com/Automattic/jetpack/assets/11078128/db0c12e3-57cf-4451-8cfe-0f814046ae8c)

.

![Markup 2024-02-12 at 15 51 50](https://github.com/Automattic/jetpack/assets/11078128/4e491382-8be2-4d96-b90f-739331eca13f)

## Proposed changes:

* Add the boost-score-api package to My Jetpack.
* Add a `latestBoostSpeedScores` property (with values), to the global `myJetpackInitialState` object.
* Add a global type definition (global.d.ts) in My Jetpack for global initialState objects.
* Add the Boost speed score into the Boost product card within My Jetpack.

**Still TODO:** (In separate follow-up PR(s)):
- Adjust time/behavior of how long we use cached previous scores. (How often we pull new speed scores. It may vary vs when the Plugin is installed & active vs when it's not.)
- Show previous scores when plugin is active.
- Create & show a ToolTip for the Boost score letter grade.


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Checkout this branch using the Jetpack Beta plugin, Or using your local dev environment.
* Not connected yet, go to My Jetpack page (`/wp-admin/admin.php?page=my-jetpack`).
* Examine the Boost product card. Verify you do not see the Boost speed score & score bar. Verify the CTA reads, "Boost your site".
* Connect Jetpack.
* Verify the Boost score and score bar looks good and appears to work properly:
    - On a brand new site, on first load, a spinner will show for a few seconds while the boost speed scores are loaded and cached.
    - After first load, the score will load quickly from cache.
